include/bringauto/fleet_protocol/http_client/FleetApiClient.hpp

```diff
@@ -0,0 +1,103 @@
+     * @param deviceName name of device
+     */
+    void setDeviceIdentification(int32_t moduleId, int32_t deviceType, std::string deviceRole, std::string deviceName);
+
+    /**
+     * @brief Calls the GET function on /cars of Fleet v2 HTTP API
+     * @return Vector of shared pointers to the Car model 
+     */
+    std::vector<std::shared_ptr<org::openapitools::client::model::Car>> getCars();
+
+    /**
+     * @brief Calls the GET function on /command/{company_name}/{car_name} of Fleet v2 HTTP API
+     * @param since optional, minimal timestamp of commands to look for
+     * @param wait optional, if true, waits for a predefined period until any command is found, will check request rate and possibly delay them
+     * @return Vector of shared pointers to the Message model containing commands in payload data
+     */
+    std::vector<std::shared_ptr<org::openapitools::client::model::Message>> getCommands(int64_t since = 0, bool wait = false);
+
+    /**
+     * @brief Calls the GET function on /status/{company_name}/{car_name} of Fleet v2 HTTP API
+     * @param since optional, minimal timestamp of statuses to look for
+     * @param wait optional, if true, waits for a predefined period until any status is found, will check request rate and possibly delay them
+     * @return Vector of shared pointers to the Message model containing statuses in payload data
+     */
+    std::vector<std::shared_ptr<org::openapitools::client::model::Message>> getStatuses(int64_t since = 0, bool wait = false);
+
+    /**
+     * Calls the POST function on /command/{company_name}/{car_name} of Fleet v2 HTTP API.
+     * setDeviceIdentification needs to be used beforehand to set DeviceId, otherwise placeholder values will be used.
+     * @param commandJson payload data of a command represented by a json as a string
+     */
+    void sendCommand(std::string commandJson);
+
+    /**
+     * Calls the POST function on /status/{company_name}/{car_name} of Fleet v2 HTTP API.
+     * setDeviceIdentification needs to be used beforehand to set DeviceId, otherwise placeholder values will be used.
+     * @param statusJson payload data of a status represented by a json as a string
+     */
+    void sendStatus(std::string statusJson);
+
+    /**
+     * @brief Calls the GET function on /available-devices/{company_name}/{car_name} of Fleet v2 HTTP API
+     * @param moduleId optional, filters returned devices to only those with matching module Id
+     * @return Shared pointer to the AvailableDevices model
+     */
+    //TODO might not work properly with api because it probably returns an array
+    std::shared_ptr<org::openapitools::client::model::AvailableDevices> getAvailableDevices(std::optional<int32_t> moduleId);
+
+private:
+    std::shared_ptr<RequestFrequencyGuard> requestFrequencyGuard;
+
+    std::shared_ptr<org::openapitools::client::api::ApiClient> apiClientPtr;
+    std::shared_ptr<org::openapitools::client::api::CarApi> carApi;
```

Comment:
car API can be unique pointer, what about others? are all of these pointers shared or only some of them>

Reply:
changed relevant pointers to unique